### PR TITLE
Add schema validation for Special Message Actions

### LINKED ISSUE #64: Update schema for X-MAN Special Message Actions
Bugzilla bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1633368

Schema: https://phabricator.services.mozilla.com/rMOZILLACENTRAL2586af74cbf90dc42bbb2cc298dd7c1d0cb71bc8#change-dJAnfcshRskP

### PR DESCRIPTION
**Pull request by ncloudioj**
This adds the schema validation for Special Message Actions defined in [here](https://phabricator.services.mozilla.com/D72633#change-dJAnfcshRskP).

Note:

* A new action was added for the `CANCEL` action
* This currently only validates the actions in `messaging-experiments.json`. There are some schema violations in `cfr.json`, which we might want to fix and enable the validation later
* It refactors the x-man experiment validation so that it only validates the branch against the inferred message schema
* It also changes the argument type for `validate.py`, such that only the schema type is required other than the schema path  

This fixes #64.

### CODE REVIEW

File: scripts/validate.py [hunk 1]
```diff
@@ -43,81 +68,110 @@ def validate_item_targeting(item, for_exp=False):
...
-        "schema/cfr-heartbeat.schema.json",
-        "schema/messaging-experiments.schema.json",
-        "schema/whats-new-panel.schema.json",
-    ]
-    for path in possible_schemas:
-        with open(path, "r") as f:
-            ALL_SCHEMAS[path] = json.loads(f.read())
+def extract_actions(message, message_type):
+    def _extract_cfr():
+        for name, button in message["content"].get("buttons").items():
```
Comment (piatra) [inline scripts/validate.py:98]: We mix moments and cfr in the same file so this will need to be guarded somehow.

**Comment by piatra** [comment]
Why are there no checks on this PR 🤔

File: scripts/validate.py (hunk 1 above)
Comment (ncloudioj) [reply scripts/validate.py:98]: Good point.

**Comment by ncloudioj** [comment]
> Why are there no checks on this PR 🤔

Yes, the CI has been weird lately, let me check if anything gets misconfigured.

**Comment by ncloudioj** [comment]
> Re the schema issues with `cfr` could it be that the `null` in the JSON gets parsed as `None` in python and then fails validation? I've tried changing the schema to `string` and `null` to `""` and everything validates.

Indeed, updating the schema for `url` to be ["string", "null"] fixed this. Will enable the action validation for `cfr.json`.

**Comment by ncloudioj** [comment]
> > Why are there no checks on this PR 🤔
> 
> Yes, the CI has been weird lately, let me check if anything gets misconfigured.

Interesting, by default CircleCI will not trigger build for forked pull requests. It's now enabled.

**Pull request merged**